polus-basic-flatfield-correction-plugin/src/basic.py

```diff
@@ -124,82 +121,82 @@ def _inexact_alm_l1(D,options):
     https://arxiv.org/pdf/1009.5055.pdf
 
     Inputs:
-        D - Stack of images used to estimate the flatfield
+        imgf_stack - Stack of images used to estimate the flatfield
         options - optimization options
     Outputs:
         new_options - Modified options
     """
     # Get basic image information and reshape input
-    p = D.shape[0]
-    q = D.shape[1]
-    m = p*q
-    n = D.shape[2]
-    D = np.reshape(D,(m,n))
-    options['weight'] = np.reshape(options['weight'],D.shape)
-    
+    img_width = imgf_stack.shape[0]
+    img_height = imgf_stack.shape[1]
+    img_size = img_width* img_height
+    img_color = imgf_stack.shape[2]
```

Comment:
This dimension is not color. Each element in the third dimension refers to a different image.